### PR TITLE
Add class AutoReturnReviewItem to return old review items to submitters

### PR DESCRIPTION
**Pull request by daisieh**
Implements https://trello.com/c/mE0kCHf0/572-process-for-returning-old-items-in-review

Test by running `dspace purge-review-item -a` and observing that items older than the amount set in settings.xml are returned to workspace.

### CODE REVIEW

File: dspace/modules/api/src/main/java/org/dspace/workflow/AutoReturnReviewItem.java [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+        options.addOption("a", "all", false, "Process all review items");
+        options.addOption("h", "help", false, "help");
+
+        setOlderThanDate();
+
+        CommandLine line = parser.parse(options, args);
+        if (line.hasOption('h'))
+        {
+            HelpFormatter myhelp = new HelpFormatter();
+            myhelp.printHelp("ItemImport\n", options);
```
Comment (ryscher) [inline dspace/modules/api/src/main/java/org/dspace/workflow/AutoReturnReviewItem.java]: Looks like the "ItemImport" is a holdover from another class...

File: dspace/modules/api/src/main/java/org/dspace/workflow/AutoReturnReviewItem.java [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+        try {
+            if (wfi != null) {
+                // make sure that this item is updated according to the ApproveReject mechanism:
+                ApproveRejectReviewItem.reviewItem(wfi);
+                claimedTasks = ClaimedTask.findByWorkflowId(context, wfi.getID());
+                //Check for a valid task
+                // There must be a claimedTask & it must be in the review stage, else it isn't a review workflowitem
+                Item item = wfi.getItem();
+                if (claimedTasks == null || claimedTasks.isEmpty() || !claimedTasks.get(0).getActionID().equals("reviewAction")) {
+                    log.debug("Item " + item.getID() + " not found or not in review");
```
Comment (ryscher) [inline dspace/modules/api/src/main/java/org/dspace/workflow/AutoReturnReviewItem.java:117]: This would be better with log.error

Comment (daisieh) [reply dspace/modules/api/src/main/java/org/dspace/workflow/AutoReturnReviewItem.java:117]: I don't know that that is true: it's not necessarily an error if something got caught and processed by ApproveRejectReviewItem earlier.

File: dspace/modules/api/src/main/java/org/dspace/workflow/AutoReturnReviewItem.java (hunk 1 above)
Comment (daisieh) [reply dspace/modules/api/src/main/java/org/dspace/workflow/AutoReturnReviewItem.java]: Good catch; it was actually a holdover in the class I copied this from, so I updated the messaging in both this class and ApproveRejectReviewItem.

**Comment by ryscher** [comment]
This tests ok on my VM. 

However, when an item is returned, the solr index is not updated, so the Wokflow Overview page will still show the item in review until the item is re-indexed.

**Comment by daisieh** [comment]
Is it important that the review item get reindexed immediately, or is it sufficient that all items modified each day are reindexed daily?

**Comment by ryscher** [comment]
Oh, that's right. I forgot about the daily reindex. 

Merging.

**Pull request merged**